hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterJoinManager.java

```diff
@@ -685,6 +704,24 @@ private boolean checkIfJoinRequestFromAnExistingMember(JoinMessage joinMessage,
         return true;
     }
 
+    private boolean isMemberRestartingWithPersistence(Member member) {
+        return member.getAttributes().get(PERSISTENCE_ENABLED_ATTRIBUTE) != null
+                && member.getAttributes().get(PERSISTENCE_ENABLED_ATTRIBUTE).equals("true")
+                // may be already detected as crashed member or probably it is still in member list because
+                // connection timeout hasn't been reached yet
+                && (hasMemberLeft(member.getUuid())
+                    || clusterService.getMembershipManager().getMember(member.getUuid()) != null);
+    }
+
+    private boolean isMemberRestartingWithPersistence(MemberInfo member) {
+        return member.getAttributes().get(PERSISTENCE_ENABLED_ATTRIBUTE) != null
+                && member.getAttributes().get(PERSISTENCE_ENABLED_ATTRIBUTE).equals("true")
+                // may be already detected as crashed member or probably it is still in member list because
+                // connection timeout hasn't been reached yet
+                && (hasMemberLeft(member.getUuid())
+                    || clusterService.getMembershipManager().getMember(member.getUuid()) != null);
+    }
```

Comment:
Both methods seem similar. Maybe we can make a common method with a signature like this:
`isMemberRestartingWithPersistence(uuid, attributes)`